README.md

```diff
@@ -2,44 +2,54 @@
 
-The library is a re-write of [polkadot staking miner](https://github.com/paritytech/polkadot/tree/master/utils/staking-miner) using [subxt](https://github.com/paritytech/subxt)
-to avoid hard dependency to each runtime version. It's using a static metadata (metadata.scale) file to generate rust types, instructions how to update metadata can be found [here](#update-metadata)
+The binary itself embeds [static metadata](./artifacts/metadata.scale) to generate a rust codegen at compile-time that [subxt provides](https://github.com/paritytech/subxt).
 
-The intention is that this library will "only break" once the [pallet-election-provider-multi-phase](https://crates.parity.io/pallet_election_provider_multi_phase/index.html) breaks i.e, not on every runtime upgrade.
+Runtime upgrades are handled by staking-miner-v2 by upgrading storage constants and that will work unless there is a breaking change in any of pallets used by the staking-miner (mainly pallet-election-provider and pallet-system are used).
 
+Because detecting breaking changes when connection to RPC node when using `staking-miner-v2` is hard, this repo performs daily integration tests against `polkadot master` and in most cases updating the metadata [see update metadata](#update-metadata)) and fixing the compile errors are sufficient.
 
+Each release will specify which runtime version it was tested against but
+it's not possible to know in advance which runtimes it will work with.
+
+Thus, it's important to subscribe to releases to this repo or add some logic that triggers an alert once the staking-miner-v2 crashes.
+
+## Usage
 
 You can check the help with:
+
 ```
 staking-miner --help
 ```
 
-## Update metadata
+### Monitor
 
-The static metadata file is stored at [artifacts/metadata.scale](artifacts/metadata.scale)
+To "mine solutions" and earn rewards, please try this on a dev-chain before using real funds because it's possible to loose money.
 
-To update the metadata you need to connect to a polkadot, kusama or westend node
+`$ cargo run --release -- --uri ws://localhost:9944 monitor --seed-or-path //Alice --dry-run seq-phragmen`
 
-```bash
-# Install subxt-cli
-$ cargo install --locked subxt-cli
-# Download the metadata from a local node and replace the current metadata
-# See `https://github.com/paritytech/subxt/tree/master/cli` for further documentation of the `subxt-cli` tool.
-$ subxt metadata -f bytes > artifacts/metadata.scale
-# Inspect the generated code
-$ subxt codegen --file artifacts/metadata.scale | rustfmt +nightly > code.rs
-```
+This is a starting point that will try compute new solutions to the validator set that also validates that
+transaction before submitting.
 
+For further information regarding the different options run:
 
-## Building
+`cargo run --release -- monitor --help`
 
-```
-cargo build --release --locked
-```
+### Dry run
+
+It's possible to mine a solution locally without submitting anything to the chain but
```

Comment:
Out of context of this PR, but in [substrate-etl](https://github.com/gpestana/substrate-etl) we pull all the data from EPM and staking and re-calculate the snapshot if it does not exist. I wonder if that would be useful here or just over-engineering and not used.

Reply:
I added a link under "related projects" instead.

It's in general "dry-running" here is tricky but using `substrate-etl` won't work unless we had some glue code to feed in the "snapshot" or just fetching the state ourselves.

I reckon that the easiest would just be to find the event in from subscan etc but open to other ideas ofc :)